### PR TITLE
Test with GHC 9.2.4 and 9.4.2 in CI

### PR DESCRIPTION
**Pull request by jhrcek**
Hello @alexbiehl 
I'd like to use this library in our project but at the moment it doesn't build with when using ghc 9.2.4 and more recent versions of Aeson. Could you please accept this PR to help me fix that?

Note that I overhauled ci config to test latest patch version of each recent minor version of GHC.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,24 +13,18 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest] # [macOS-latest, windows-latest] do not come with docker :(
-        cabal: ["3.2"]
+        cabal: ["3.6"]
         ghc:
-          - "8.8.3"
-          - "8.10.1"
-          - "8.10.2"
-        exclude:
```
Comment (jhrcek) [inline .github/workflows/ci.yaml:21]: These excludes were pointless, as you don't test with macOS-latest and windows-latest anyway.

**Review by alexbiehl — APPROVED** [review]
Sweet thanks!

**Comment by jhrcek** [comment]
Actually please wait a bit with merging, I figured we could take advantage of IsString typeclass that both Text and Aeson.Key implement and that way get rid of all the CPP stuff. Also I just tried and it seems to work with ghc 9.4.2, so why not add that to CI as well.

**Comment by jhrcek** [comment]
All GHCs passed in CI in my fork: https://github.com/jhrcek/testcontainers-hs/actions/runs/3320353421/jobs/5486631895
I consider it ready to merge.

File: src/TestContainers/Docker.hs [hunk 2]
```diff
@@ -1155,8 +1156,9 @@ containerPort :: Container -> Int -> Int
 containerPort Container { id, inspectOutput } port =
   let
     -- TODO also support UDP ports
-    textPort :: Text
-    textPort = pack (show port) <> "/tcp"
+    -- Using IsString so it works both with Text (aeson<2) and Aeson.Key (aeson>=2)
```
Comment (alexbiehl) [inline src/TestContainers/Docker.hs:1159]: Thanks for the clarifying comment!

**Comment by alexbiehl** [comment]
@jhrcek is your usage of the library open source? Would it be possible for you to link to your repo/tests for an example?

**Comment by jhrcek** [comment]
Unfortunately it's not. It lives as private repo under https://github.com/Holmusk/

**Comment by jhrcek** [comment]
Hello @alexbiehl 
ould you please release ghc-9.2.x supporting version to hackage?
We're already using master branch of this repo with 9.2.5, but we'd like to switch to hackage dependency instead.

**Pull request merged**